v2/observer/jobs.go

```diff
@@ -176,3 +191,67 @@ func (o *observer) deleteJobResources(
 		)
 	}
 }
+
+func (o *observer) startJobPodTimer(ctx context.Context, pod *corev1.Pod) {
+	if pod.Status.Phase != corev1.PodPending &&
+		pod.Status.Phase != corev1.PodRunning {
+		return
+	}
+	if pod.Annotations["timeoutSeconds"] == "" {
+		return
+	}
+
+	duration := pod.Annotations["timeoutSeconds"] + "s"
+	timeout, err := time.ParseDuration(duration)
+	if err != nil {
+		o.errFn(
+			errors.Wrapf(
+				err,
+				"unable to parse timeout duration %q for pod %q",
+				duration,
+				pod.Name,
+			),
+		)
+		return
+	}
+
+	go func() {
+		ticker := time.NewTicker(timeout)
+		defer ticker.Stop()
```

Comment:
I'd suggest a `time.Timer` over a `time.Ticker`. The timer goes off once like an alarm clock instead of repeatedly at intervals.